presto-main/src/main/java/com/facebook/presto/metadata/FunctionManager.java

```diff
@@ -174,6 +174,15 @@ public void createFunction(SqlInvokedFunction function, boolean replace)
         functionNamespaceManager.get().createFunction(function, replace);
     }
 
+    public void dropFunction(QualifiedFunctionName functionName, Optional<List<TypeSignature>> parameterTypes, boolean exists)
+    {
+        Optional<FunctionNamespaceManager<?>> functionNamespaceManager = getServingFunctionNamespaceManager(functionName.getFunctionNamespace());
+        if (!functionNamespaceManager.isPresent()) {
+            throw new PrestoException(GENERIC_USER_ERROR, format("Cannot drop function in function namespace: %s", functionName.getFunctionNamespace()));
```

Comment:
Should the error message be "cannot find function"? The reason we cannot drop is because we do not have a function manager that serve the given function, which means that function does not exist. So if the `exists` flag is set, this can just succeed. Otherwise it should be a `FUNCTION_NOT_FOUND`?